File: sdk/csharp/src/Hjtpx.Captcha.Sdk/Client/CaptchaClientConfig.cs
using Hjtpx.Captcha.Pool;
using Hjtpx.Captcha.Retry;

namespace Hjtpx.Captcha.Client;

public class CaptchaClientConfig
{
    public string BaseUrl { get; set; } = string.Empty;
    public string? ApiKey { get; set; }
    public string? SecretKey { get; set; }
    public ConnectionPoolConfig ConnectionPoolConfig { get; set; } = new ConnectionPoolConfig();
    public RetryConfig RetryConfig { get; set; } = new RetryConfig();

    public CaptchaClientConfig()
    {
    }

    public CaptchaClientConfig(string baseUrl)
    {
        BaseUrl = baseUrl;
    }

    public CaptchaClientConfig(string baseUrl, string apiKey)
    {
        BaseUrl = baseUrl;
        ApiKey = apiKey;
    }

    public CaptchaClientConfig(string baseUrl, string apiKey, string secretKey)
    {
        BaseUrl = baseUrl;
        ApiKey = apiKey;
        SecretKey = secretKey;
    }
}

File: sdk/java/src/main/java/com/hjtpx/captcha/client/CaptchaClientConfig.java
package com.hjtpx.captcha.client;

import com.hjtpx.captcha.pool.ConnectionPoolConfig;
import com.hjtpx.captcha.retry.RetryConfig;

public class CaptchaClientConfig {
    private String baseUrl;
    private String apiKey;
    private String secretKey;
    private ConnectionPoolConfig connectionPoolConfig;
    private RetryConfig retryConfig;

    public CaptchaClientConfig() {
        this.connectionPoolConfig = new ConnectionPoolConfig();
        this.retryConfig = new RetryConfig();
    }

    public CaptchaClientConfig(String baseUrl) {
        this();
        this.baseUrl = baseUrl;
    }

    public CaptchaClientConfig(String baseUrl, String apiKey) {
        this(baseUrl);
        this.apiKey = apiKey;
    }

    public CaptchaClientConfig(String baseUrl, String apiKey, String secretKey) {
        this(baseUrl, apiKey);
        this.secretKey = secretKey;
    }

    public String getBaseUrl() {
        return baseUrl;
    }

    public void setBaseUrl(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    public String getApiKey() {
        return apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public String getSecretKey() {
        return secretKey;
    }

    public void setSecretKey(String secretKey) {
        this.secretKey = secretKey;
    }

    public ConnectionPoolConfig getConnectionPoolConfig() {
        return connectionPoolConfig;
    }

    public void setConnectionPoolConfig(ConnectionPoolConfig connectionPoolConfig) {
        this.connectionPoolConfig = connectionPoolConfig;
    }

    public RetryConfig getRetryConfig() {
        return retryConfig;
    }

    public void setRetryConfig(RetryConfig retryConfig) {
        this.retryConfig = retryConfig;
    }
}
